ouroboros-consensus/test/consensus-test/Test/Consensus/MiniProtocol/ChainSync/Client.hs

```diff
@@ -2,8 +2,8 @@
 {-# LANGUAGE FlexibleContexts    #-}
 {-# LANGUAGE GADTs               #-}
 {-# LANGUAGE LambdaCase          #-}
+{-# LANGUAGE NamedFieldPuns      #-}
 {-# LANGUAGE RankNTypes          #-}
-{-# LANGUAGE RecordWildCards     #-}
 {-# LANGUAGE ScopedTypeVariables #-}
 {-# LANGUAGE TypeApplications    #-}
 -- | Tests for the chain sync client.
```

Comment:
@amesgen Please confirm that this module is testing strictly _more_ than it was before. For that purpose, I split the changes into two parts. I'm hoping it's relatively trivial to see that the first commit is purely a refactor (... modulo some innocuous RTS thread scheduling, maybe?). And the second one is purely additive.

Reply:
Notably: some blocks are from the future now. But not far future, since those trivialize the test (maybe we should include those even so?).